icu4c/source/i18n/complexunitsconverter.cpp

```diff
@@ -0,0 +1,148 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+
+#include <cmath>
+
+#include "cmemory.h"
+#include "complexunitsconverter.h"
+#include "uarrsort.h"
+#include "uassert.h"
+#include "unicode/fmtable.h"
+#include "unicode/localpointer.h"
+#include "unicode/measunit.h"
+#include "unicode/measure.h"
+#include "unitconverter.h"
+
+U_NAMESPACE_BEGIN
+namespace units {
+
+ComplexUnitsConverter::ComplexUnitsConverter(const MeasureUnitImpl &inputUnit,
+                                             const MeasureUnitImpl &outputUnits,
+                                             const ConversionRates &ratesInfo, UErrorCode &status)
+    : units_(outputUnits.extractIndividualUnits(status)) {
+    if (U_FAILURE(status)) {
+        return;
+    }
+
+    U_ASSERT(units_.length() != 0);
+
+    // NOTE:
+    //  This comparator is used to short the units in a descending order. Therefore, we return -1 if
```

Comment:
sort not short

Reply:
Done.

Reply:
(Commit 41d7ee071e270a030af5091f6e0fd8de197cbb87, found in the history of tag "hugovdm_upstreaming_step20".)